src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Unix.cs

```diff
@@ -102,6 +100,19 @@ private static void CreateFile(string folderPath, string permissions)
             File.WriteAllText(filename, "contents");
 
             Assert.Equal(0, Interop.Sys.ChMod(filename, Convert.ToInt32(permissions, 8)));
+
+            // In some environments, the file mode may be modified by the OS.
+            // See the Rationale section of https://linux.die.net/man/3/chmod.
+
+            // To workaround this, read the file mode back, and if it has changed rename the file.
+            Interop.Sys.FileStatus status;
+            Assert.Equal(0, Interop.Sys.Stat(filename, out status));
+            string updatedPermissions = Convert.ToString(status.Mode & 0xFFF, 8);
+            if (updatedPermissions != permissions)
+            {
+                string newFileName = Path.Combine(folderPath, $"{updatedPermissions}.txt");
```

Comment:
What is the purpose of renaming?

Reply:
The expected file mode is encoded in the file name. So the file named `755.txt` expects the file mode to be `755`.

See https://github.com/dotnet/runtime/blob/334989f6cc0600f9c12e9fa989dc1814c8300d1d/src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Unix.cs#L36-L40